loopd/swapclient_server.go

```diff
@@ -135,19 +135,34 @@ func (s *swapClientServer) marshallSwap(loopSwap *loop.SwapInfo) (
 		return nil, errors.New("unknown swap type")
 	}
 
-	return &looprpc.SwapStatus{
+	status := &looprpc.SwapStatus{
 		Amt:            int64(loopSwap.AmountRequested),
 		Id:             loopSwap.SwapHash.String(),
 		IdBytes:        loopSwap.SwapHash[:],
 		State:          state,
 		InitiationTime: loopSwap.InitiationTime.UnixNano(),
 		LastUpdateTime: loopSwap.LastUpdate.UnixNano(),
-		HtlcAddress:    loopSwap.HtlcAddress.EncodeAddress(),
 		Type:           swapType,
 		CostServer:     int64(loopSwap.Cost.Server),
 		CostOnchain:    int64(loopSwap.Cost.Onchain),
 		CostOffchain:   int64(loopSwap.Cost.Offchain),
-	}, nil
+	}
+
+	if loopSwap.SwapType == swap.TypeIn {
```

Comment:
Not sure if this matters, but what happens if the client has an internal loop in with NP2WSH on the go then upgrades their client? We're expecting nested htlcs only to be for external htlcs, but there is an in the middle upgrade case where an internal swap will still have a nested htlc.